File: library/src/main/java/no/wtw/android/restserviceutils/task/BackgroundTask.java
package no.wtw.android.restserviceutils.task;

import android.content.Context;

public interface BackgroundTask<D> {

    void onLoadStart();

    D onLoadExecute() throws Exception;

    void onLoadSuccess(D data);

    void onLoadError(Context context, Exception e);

    void onLoadEnd(Context context);

}

File: library/src/main/java/no/wtw/android/restserviceutils/task/BackgroundTask.kt
package no.wtw.android.restserviceutils.task

interface BackgroundTask<D> {
    fun onLoadStart()

    @Throws(Exception::class)
    fun onLoadExecute(): D

    fun onLoadSuccess(data: D)

    fun onLoadError(e: Exception)

    fun onLoadEnd()

}